middleware/verify_privilege_token/main.go

```diff
@@ -0,0 +1,64 @@
+package verify_privilege_token
+
+import (
+	"fmt"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/rs/zerolog"
+	"golang.org/x/exp/slices"
+)
+
+type VerifyPrivilegeTokenInterface interface {
+	HasTokenPrivilege(privilegeID int64) fiber.Handler
+}
+
+type Config struct {
+	Log *zerolog.Logger
+}
+
+type verifyPrivilegeToken struct {
+	cfg Config
+}
+
+func New(cfg Config) VerifyPrivilegeTokenInterface {
+	return &verifyPrivilegeToken{
+		cfg: cfg,
+	}
+}
+
+func (p *verifyPrivilegeToken) HasTokenPrivilege(privilegeID int64) fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		return p.checkPrivilege(c, privilegeID)
+	}
+}
+
+func (p *verifyPrivilegeToken) checkPrivilege(c *fiber.Ctx, privilegeID int64) error {
+	logger := p.cfg.Log
+	claims, err := getVehicleTokenClaims(c)
+	if err != nil {
+		logger.Debug().Str("VehicleTokenID In Request", c.Params("tokenID")).
+			Str("VehicleTokenID in bearer token", claims.VehicleTokenID).
+			Msg(err.Error())
+		return fiber.NewError(fiber.StatusInternalServerError, "Error verifying user privilege!")
```

Comment:
This is probably a 401, yeah? What would you return if someone passed in a mangled identity token?

Reply:
hmmm.. i might need to look into how to get the specific error then, since this can also be caused (on rare occassions) by the server not being able to decode the jwt token.